lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -82,12 +90,14 @@
     private final TupleDomain<ColumnDescriptor> effectivePredicate;
     private final List<ColumnDescriptor> columns;
     private final DateTimeZone timeZone;
+    private final int domainCompactionThreshold;
 
-    public TupleDomainParquetPredicate(TupleDomain<ColumnDescriptor> effectivePredicate, List<ColumnDescriptor> columns, DateTimeZone timeZone)
+    public TupleDomainParquetPredicate(TupleDomain<ColumnDescriptor> effectivePredicate, List<ColumnDescriptor> columns, DateTimeZone timeZone, int domainCompactionThreshold)
```

Comment:
`domainCompactionThreshold` is used only in `matches(BloomFilterStore)` method.
What speaks against passing it directly to that method?

Reply:
this is applied in a fixup now, it helps somewhat but the param still needs to come from connector classes